server/hotrod/src/test/java/org/infinispan/server/hotrod/transport/TestHandlersChannelInitializer.java

```diff
@@ -17,11 +17,16 @@
  * @since 9.0
  */
 public class TestHandlersChannelInitializer implements NettyInitializer {
+   public static final String DISABLE_FIXED_DECODER = "infinispan.test.server.disable-fixed";
+
+   private final boolean disableFixedDecoder = Boolean.parseBoolean(System.getProperty(DISABLE_FIXED_DECODER, "false"));
```

Comment:
I do not like this as it will leak into all the other tests. I think we need to add an additional override to HotRodTestingUtil#startHotRodServer to prevent using the test channel initializer. Then each test can control if this is applied or not.

Reply:
Updated to just add a new method instead.